src/measure_theory/measure/complex.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+
+import measure_theory.decomposition.jordan
+
+/-!
+# Complex measure
+
+This file proves some elementary results about complex measures. In particular, we prove that
+a complex measure is always in the form `s + it` where `s` and `t` are signed measures.
+
+## Main definitions
+
+* `measure_theory.complex_measure.re_part`: obtains a signed measure `s` from a complex measure `c`
+  such that `s i = (c i).re` for all measurable sets `i`.
+* `measure_theory.complex_measure.im_part`: obtains a signed measure `s` from a complex measure `c`
+  such that `s i = (c i).im` for all measurable sets `i`.
+* `measure_theory.signed_measure.to_complex_measure`: given two signed measures `s` and `t`,
+  `s.to_complex_measure t` provides a complex measure of the form `s + it`.
+* `measure_theory.complex_measure.equiv_signed_measure`: is the equivalence between the complex
+  measures and the type of the product of the signed measures with itself.
+
+# Tags
+
+Complex measure
+-/
+
+noncomputable theory
+open_locale classical measure_theory ennreal nnreal
+
+variables {α β : Type*} {m : measurable_space α}
+
+namespace measure_theory
+
+open vector_measure
+
+namespace complex_measure
+
+include m
+
+/-- The real part of a complex measure is a signed measure. -/
+@[simps]
+def re_part (c : complex_measure α) : signed_measure α :=
+{ measure_of' := λ i, (c i).re,
+  empty' := by simp only [c.empty, complex.zero_re],
+  not_measurable' := λ i hi, by simp only [c.not_measurable hi, complex.zero_re],
+  m_Union' := λ f hfm hfdisj, (c.m_Union hfm hfdisj).mapL complex.re_clm }
+
+lemma re_part_add (c d : complex_measure α) :
+  (c + d).re_part = c.re_part + d.re_part :=
+rfl
+
+/-- The imaginary part of a complex measure is a signed measure. -/
+@[simps]
+def im_part (c : complex_measure α) : signed_measure α :=
+{ measure_of' := λ i, (c i).im,
+  empty' := by simp only [c.empty, complex.zero_im],
+  not_measurable' := λ i hi, by simp only [c.not_measurable hi, complex.zero_im],
+  m_Union' := λ f hfm hfdisj, (c.m_Union hfm hfdisj).mapL complex.im_clm }
```

Comment:
Is there a `vector_measure.map` missing here, such that this could just be:
```suggestion
c.map complex.im_clm
```

Reply:
We do have a `vector_measure.map` but that's a different thing. I think the version that would work here would not work in general since a (measurable)-function might map a measurable set to an unmeasurable one.

Reply:
I don't see which part of the proof is using some special property of `complex.im_clm` that doesn't hold in general.

Reply:
You are right, I was thinking about properties required to make vector_measure.map work. I can make a new definition similar to what you said. What do you think we should name it?

Reply:
`map_range` maybe?

Reply:
I've added it now